compiler/luci/profile/include/luci/Profile/CircleNodeOrigin.h

```diff
@@ -32,7 +32,7 @@ class CircleNodeOrigin
   struct Source
   {
   public:
-    std::string name(void) const { return _name; }
+    const std::string& name(void) const { return _name; }
```

Comment:
plz do not change to `const`

Reply:
Does this introduce an API break or is this return-by-copy intended? This particular change is an optimisation in a way because it avoids unnecessary copies at the call site.

Reply:
I want to leave it as-is to make variable writable.

Reply:
The writability of the returned value is controlled by the cv-specifiers of the variable itself, not the return value specification. If the return-by-value is kept we get an unnecessary copy if the caller actually needs to refer to the name but doesn't necessarily need a copy of this internal string.
```
Source s;
std::string name1 = s.name();
const std::string name2 = s.name();
const std::string& name2 = s.name(); // unnecessary copy avoided
```
An example of such situation is looking for a source by name:
```
const luci::Node node = (...);
const auto origin = luci::get_origin(node);
for (const auto* source : origin->sources()) {
    if (source->name() == "some_name") { // the returned string doesn't have to be a copy
        // do something with this source
    }
}
```
and this is exactly what I encountered in the NPU compiler which depends on this code.